lib/saml/user.rb

```diff
@@ -10,113 +10,89 @@ module SAML
       %i[authn_context]
     end
 
-    def dslogon?
-      saml_attributes.to_h.keys.include?('dslogon_uuid')
-    end
+    # NOTE: The actual exception, if any, should get raised when to_hash is called. Hence "suppress"
+    def log_warnings_to_sentry
+      user_attributes.to_hash
 
-    def mhv?
-      saml_attributes.to_h.keys.include?('mhv_uuid')
-    end
+      if user_attributes.warnings.any?
+        warning_context = {
+          authn_context: authn_context,
+          warnings: user_attributes.warnings.uniq.join(', ')
+        }
 
-    # see warnings
-    # NOTE: The actual exception, if any, should get raised when to_hash is called. Hence "suppress"
-    def log_warnings_to_sentry!
-      suppress(Exception) do
-        if (warnings = warnings_for_sentry).any?
-          warning_context = {
-            real_authn_context: real_authn_context,
-            authn_context: authn_context,
-            warnings: warnings.join(', '),
-            loa: user_attributes.loa
-          }
-          log_message_to_sentry("Issues in SAML Response - #{real_authn_context}", :warn, warning_context)
-        end
+        log_message_to_sentry('SAML RESPONSE WARNINGS', :warn, warning_context)
       end
     end
 
-    # will be one of [loa1, loa3, multifactor, dslogon, mhv]
+    # will be one of AUTHN_CONTEXTS.keys
     # this is the real authn-context returned in the response without the use of heuristics
-    def real_authn_context
+    def authn_context
       REXML::XPath.first(saml_response.decrypted_document, '//saml:AuthnContextClassRef')&.text
-    # this is to add additional context when we cannot parse for authn_context
-    rescue NoMethodError
-      Raven.extra_context(
-        base64encodedpayload: Base64.encode64(saml_response.response),
-        attributes: saml_response.attributes.to_h
-      )
+    rescue StandardError
       Raven.tags_context(controller_name: 'sessions', sign_in_method: 'not-signed-in:error')
       raise
     end
 
-    # We want to do some logging of when and how the following issues could arise, since loa is
-    # derived based on combination of these values, it could raise an exception at any time, hence
-    # why we use try/catch.
-    def warnings_for_sentry
-      warnings = []
-      warnings << 'LOA Current Nil' if user_attributes.loa_current.blank?
-      warnings << 'LOA Highest Nil' if user_attributes.loa_highest.blank?
-      warnings
-    end
-
     # should eventually have a special case for multifactor policy and refactor all of this
     # but session controller refactor is premature and can't handle it right now.
     def user_attributes_class
       case authn_context
       when 'myhealthevet'; then SAML::UserAttributes::MHV
       when 'dslogon'; then SAML::UserAttributes::DSLogon
-      else
+      when 'myhealthevet_multifactor', 'dslogon_multifactor', 'multifactor'
         SAML::UserAttributes::IdMe
+      when 'dslogon_loa3', 'myhealthevet_loa3', LOA::IDME_LOA3
```

Comment:
simplify to?
````ruby 
      when 'myhealthevet_multifactor', 'dslogon_multifactor', 'multifactor',
            'dslogon_loa3', 'myhealthevet_loa3', LOA::IDME_LOA3,
             LOA::IDME_LOA1
        SAML::UserAttributes::IdMe
````

Reply:
yes, the line was too long and I didn't like the way it looked with the when clause spanning multiple lines. 😂